test/suites/termination/termination_grace_period_test.go

```diff
@@ -55,19 +54,28 @@ var _ = Describe("TerminationGracePeriod", func() {
 				return karpv1.IsDisruptingTaint(t)
 			})
 			g.Expect(ok).To(BeTrue())
-		}).Should(Succeed())
+		}).WithTimeout(3 * time.Second).WithPolling(100 * time.Millisecond).Should(Succeed())
+
+		// Check that pod remains healthy until termination grace period
+		// subtract the polling time of the eventually above to reduce any races.
```

Comment:
It looks like this currently assumes we'll be successful within a single polling iteration, I'm wondering how often that's true. Could we rely on the TGP annotation instead to determine how long the pod should remain healthy?

Reply:
Ah got it, makes sense. SGTM